src/rendering/framebufferrenderer.cpp

```diff
@@ -1206,169 +1194,148 @@ void FramebufferRenderer::render(float blackoutFactor, bool doPerformanceMeasure
-    _camera = camera;
-}
+    void FramebufferRenderer::setHDRExposure(float hdrExposure) {
+        _hdrExposure = hdrExposure;
+        if (_hdrExposure < 0.0f) {
+            LERROR("HDR Exposure constant must be greater than zero.");
+            _hdrExposure = 1.0f;
+        }
+    }
 
-void FramebufferRenderer::setResolution(glm::ivec2 res) {
-    _resolution = res;
-    _dirtyResolution = true;
-}
+    void FramebufferRenderer::setHDRBackground(float hdrBackground) {
+        _hdrBackground = hdrBackground;
+        if (_hdrBackground < 0.0f) {
+            LERROR("HDR Background constant must be greater than zero.");
+            _hdrBackground = 1.0f;
+        }
+    }
 
-void FramebufferRenderer::setNAaSamples(int nAaSamples) {
-    _nAaSamples = nAaSamples;
-    if (_nAaSamples == 0) {
-        _nAaSamples = 1;
+    void FramebufferRenderer::setGamma(float gamma) {
+        _gamma = gamma;
+        if (_gamma < 0.0f) {
+            LERROR("Gamma value must be greater than zero.");
+            _gamma = 2.2f;
+        }
     }
-    if (_nAaSamples > 8) {
-        LERROR("Framebuffer renderer does not support more than 8 MSAA samples.");
-        _nAaSamples = 8;
+
+    float FramebufferRenderer::hdrBackground() const {
+        return _hdrBackground;
     }
-    _dirtyResolution = true;
-}
-
-void FramebufferRenderer::setHDRExposure(float hdrExposure) {
-    _hdrExposure = hdrExposure;
-    if (_hdrExposure < 0.0f) {
-        LERROR("HDR Exposure constant must be greater than zero.");
-        _hdrExposure = 1.0f;
+
+    int FramebufferRenderer::nAaSamples() const {
+        return _nAaSamples;
     }
-}
 
-void FramebufferRenderer::setHDRBackground(float hdrBackground) {
-    _hdrBackground = hdrBackground;
-    if (_hdrBackground < 0.0f) {
-        LERROR("HDR Background constant must be greater than zero.");
-        _hdrBackground = 1.0f;
+    std::vector<double> FramebufferRenderer::mSSAPattern() const {
+        return _mSAAPattern;
     }
-}
 
-void FramebufferRenderer::setGamma(float gamma) {
-    _gamma = gamma;
-    if (_gamma < 0.0f) {
-        LERROR("Gamma value must be greater than zero.");
-        _gamma = 2.2f;
+    void FramebufferRenderer::updateRendererData() {
+        ghoul::Dictionary dict;
+        dict.setValue("fragmentRendererPath", std::string(RenderFragmentShaderPath));
+        _rendererData = dict;
+        OsEng.renderEngine().setRendererData(dict);
     }
-}
 
-float FramebufferRenderer::hdrBackground() const {
-    return _hdrBackground;
-}
+    void saveTextureToPPMFile(const GLenum color_buffer_attachment,
+        const std::string & fileName, const int width, const int height)
+    {
+        std::fstream ppmFile;
 
-int FramebufferRenderer::nAaSamples() const {
-    return _nAaSamples;
-}
+        ppmFile.open(fileName.c_str(), std::fstream::out);
+        if (ppmFile.is_open()) {
+            unsigned char* pixels = new unsigned char[width*height * 3];
```

Comment:
Better to use std::vector<unsigned char> in case someone throws an exception before the delete[]